test/e2e/network/netpol/network_policy.go

```diff
@@ -1088,6 +1088,47 @@ var _ = common.SIGDescribe("Netpol [LinuxOnly]", func() {
 			reachability.ExpectAllIngress(NewPodString(nsX, "a"), false)
 			ValidateOrFail(k8s, model, &TestCase{ToPort: 81, Protocol: v1.ProtocolTCP, Reachability: reachability})
 		})
+
+		ginkgo.It("should enforce policy to allow traffic based on NamespaceSelector with MatchLabels using immutable ns label [Feature:NetworkPolicy][Feature:NamespaceDefaultLabelName]", func() {
+			nsX, nsY, nsZ, model, k8s := getK8SModel(f)
+			allowedLabels := &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					v1.LabelMetadataName: nsY,
+				},
+			}
+			ingressRule := networkingv1.NetworkPolicyIngressRule{}
+			ingressRule.From = append(ingressRule.From, networkingv1.NetworkPolicyPeer{NamespaceSelector: allowedLabels})
+			policy := GenNetworkPolicyWithNameAndPodMatchLabel("allow-client-a-via-ns-selector-for-immutable-ns-label", map[string]string{"pod": "a"}, SetSpecIngressRules(ingressRule))
+			CreatePolicy(k8s, policy, nsX)
+
+			reachability := NewReachability(model.AllPods(), true)
+			reachability.ExpectPeer(&Peer{Namespace: nsX}, &Peer{Namespace: nsX, Pod: "a"}, false)
+			reachability.ExpectPeer(&Peer{Namespace: nsZ}, &Peer{Namespace: nsX, Pod: "a"}, false)
+			reachability.ExpectPeer(&Peer{Namespace: nsY}, &Peer{Namespace: nsX, Pod: "a"}, true)
+
+			ValidateOrFail(k8s, model, &TestCase{ToPort: 80, Protocol: v1.ProtocolTCP, Reachability: reachability})
+		})
+
+		ginkgo.It("should enforce policy based on NamespaceSelector with MatchExpressions using immutable ns label [Feature:NetworkPolicy][Feature:NamespaceDefaultLabelName]", func() {
```

Comment:
Can you change `immutable` to `default` in this PR? 

The key to this test is that we're testing the default labels on namespaces.